### PR TITLE
networkv2: Fix api names and functions

### PR DESCRIPTION
**Pull request by amandawrcoelho**
issue #1716

Signed-off-by: Amanda Coelho <amada.winder.ribeiro.coelho@intel.com>

### CODE REVIEW

File: src/lib/comms/sol-network-impl-contiki.c [hunk 1]
```diff
@@ -161,20 +161,20 @@ sol_network_shutdown(void)
...
 }
 
-SOL_API bool
+SOL_API int
 sol_network_subscribe_events(void (*cb)(void *data, const struct sol_network_link *link,
     enum sol_network_event event),
     const void *data)
 {
-    return false;
+    return -errno;
```
Comment (cabelitos) [inline src/lib/comms/sol-network-impl-contiki.c]: This is not quite correctly. You must return the errno value not the value of the errno variable.
What I mean, is that you need to do something like: `return -ENOSYS;`

File: src/lib/comms/sol-network-impl-contiki.c [hunk 2]
```diff
@@ -161,20 +161,20 @@ sol_network_shutdown(void)
...
 }
 
-SOL_API bool
+SOL_API int
 sol_network_unsubscribe_events(void (*cb)(void *data, const struct sol_network_link *link,
     enum sol_network_event event),
     const void *data)
 {
-    return false;
+    return -EFAULT;
```
Comment (cabelitos) [inline src/lib/comms/sol-network-impl-contiki.c]: Use -ENOSYS here

File: src/lib/comms/sol-network-impl-linux.c [hunk 3]
```diff
@@ -464,24 +464,24 @@ sol_network_subscribe_events(void (*cb)(void *data, const struct sol_network_lin
 
     SOL_NULL_CHECK(cb, false);
 
-    if (sol_network_start_netlink() < 0) return false;
+    if (sol_network_start_netlink() < 0) return -errno;
```
Comment (cabelitos) [inline src/lib/comms/sol-network-impl-linux.c]: Again, return the `errno` value is not correct in this case
I suggest that you do the following:
```C
int r;
r = sol_network_start_netlink();
SOL_INT_CHECK(r, < 0, r);
```

File: src/lib/comms/sol-network-impl-linux.c [hunk 4]
```diff
@@ -464,24 +464,24 @@ sol_network_subscribe_events(void (*cb)(void *data, const struct sol_network_lin
...
 
     callback = sol_vector_append(&network->callbacks);
     SOL_NULL_CHECK(callback, false);
```
Comment (cabelitos) [inline src/lib/comms/sol-network-impl-linux.c]: You must also change this return value. Replace `false` with `-ENOMEM`
Remenber to check all return values and replace `true` and `false` with the proper values

File: src/lib/comms/sol-network-impl-linux.c [hunk 5]
```diff
@@ -580,13 +580,13 @@ sol_network_link_set_status(uint16_t link_index, unsigned int changes, unsigned
     return true;
 }
 
-SOL_API bool
+SOL_API int
 sol_network_link_up(uint16_t link_index)
 {
     return sol_network_link_set_status(link_index, IFF_UP, IFF_UP);
```
Comment (cabelitos) [inline src/lib/comms/sol-network-impl-linux.c]: This is wrong. `sol_network_link_set_status()` returns a `bool` and not an `int`

File: src/lib/comms/sol-network-impl-linux.c [hunk 6]
```diff
@@ -580,13 +580,13 @@ sol_network_link_set_status(uint16_t link_index, unsigned int changes, unsigned
...
 }
 
-SOL_API bool
+SOL_API int
 sol_network_link_down(uint16_t link_index)
 {
     return sol_network_link_set_status(link_index, IFF_UP, ~IFF_UP);
```
Comment (cabelitos) [inline src/lib/comms/sol-network-impl-linux.c]: Same as above

File: src/lib/comms/sol-network-impl-riot.c [hunk 7]
```diff
@@ -173,20 +173,20 @@ sol_network_shutdown(void)
...
 }
 
-SOL_API bool
+SOL_API int
 sol_network_subscribe_events(void (*cb)(void *data, const struct sol_network_link *link,
     enum sol_network_event event),
     const void *data)
 {
-    return false;
+    return -errno;
```
Comment (cabelitos) [inline src/lib/comms/sol-network-impl-riot.c]: replace `-errno` with `-ENOSYS`

File: src/lib/comms/sol-network-impl-contiki.c (hunk 1 above)
Comment (ceolin) [inline src/lib/comms/sol-network-impl-contiki.c]: yes, you do not even know what is the value in the errno variable

File: src/lib/comms/sol-network-impl-linux.c (hunk 5 above)
Comment (amandawrcoelho) [inline src/lib/comms/sol-network-impl-linux.c]: In this case, should I change the sol_network_link_status( ) or should I keep it as the same way and change sol_network_link_up( ) and change sol_network_link_down( )?

Comment (cabelitos) [inline src/lib/comms/sol-network-impl-linux.c]: @amandawrcoelho You should change the functions `sol_network_link_set_status()` return value type

**Pull request closed without merging**